### PR TITLE
Cover images improvment

### PR DESCRIPTION
**Pull request by aurangzaib048**

### CODE REVIEW

File: src/favicons_covers/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+### `process_site` function
+The `process_site` function is designed to retrieve information about a given domain, specifically focusing on
+obtaining the site's cover image and background color. The function makes use of various methods to fetch this
+information, employing multiple fallback mechanisms to ensure robustness in case of failures.
+
+1. **Fetching Favicon using Google's Service:**
+    - First attempt to fetch the cover image using Google's Favicon API.
+    - If successful, retrieve the cover image and check its size. If the size is too small, raise a `ValueError`.
+    - Obtain the background color associated with the cover image.
+2. **Fallback to Alternative cover image retrieval from the Website (using selectors):**
```
Comment (LorenzoMinto) [inline src/favicons_covers/README.md:23]: Would be great to specify what these alternative selectors are

File: src/favicons_covers/color.py [hunk 2]
```diff
@@ -6,6 +6,61 @@
...
+
+    return faded_hex_color
+
+
+def is_monochromatic(image, tolerance=5):
+    colored_image = image.convert("RGBA")
+
+    image_array = np.array(colored_image)
+
+    std_rgb = [
```
Comment (LorenzoMinto) [inline src/favicons_covers/color.py:36]: we can take the non-transparent pixel indexing `image_array[:, :, 3] != 0` out and only compute it once

File: src/favicons_covers/color.py [hunk 3]
```diff
@@ -6,6 +6,61 @@
...
+        int(np.std(image_array[:, :, 0][image_array[:, :, 3] != 0])),
+        int(np.std(image_array[:, :, 1][image_array[:, :, 3] != 0])),
+        int(np.std(image_array[:, :, 2][image_array[:, :, 3] != 0])),
+    ]
+
+    count_smaller_than_threshold = len(
+        [value for value in std_rgb if value < tolerance]
+    )
+
+    return count_smaller_than_threshold >= 1
```
Comment (LorenzoMinto) [inline src/favicons_covers/color.py:46]: this is saying: `at least 1 of the channels is constant`, I'm not sure why we are not checking that all three channels are constant (which should be by definition a monochromatic image). Do you have an intuition/explanation around this?

File: src/favicons_covers/color.py [hunk 4]
```diff
@@ -6,6 +6,61 @@
...
+
+
+def has_transparency(image):
```
Comment (LorenzoMinto) [inline src/favicons_covers/color.py:49]: If I'm reading the code correctly, image input is always RGBA (we only get images through `get_icon`), why are we checking for the other modes?

File: src/favicons_covers/cover_images.py [hunk 5]
```diff
@@ -188,6 +203,10 @@ def get_background_color(image: Image):
     the median edge color. That is, the middle most color of all
     the edge pixels in the image.
     """
+    if has_transparency(image):
```
Comment (LorenzoMinto) [inline src/favicons_covers/cover_images.py:206]: would be good to add a comment that if it has transparency and it is monochromatic then it's likely to be a logo. But what if the logo is not monochromatic?

Comment (aurangzaib048) [reply src/favicons_covers/cover_images.py:206]: Yeah, once we will get expected results, I will update the read me as well as the docs strings.

File: src/favicons_covers/color.py (hunk 4 above)
Comment (aurangzaib048) [reply src/favicons_covers/color.py:49]: Yeah, we can only check the last condition but I got it from here: https://stackoverflow.com/questions/43864101/python-pil-check-if-image-is-transparent, which checks all the possibilities.

File: src/favicons_covers/color.py (hunk 3 above)
Comment (aurangzaib048) [reply src/favicons_covers/color.py:46]: I have to test with larger thresholds, than I can with the detail explanations and edge cases.

File: src/favicons_covers/color.py (hunk 2 above)
Comment (LorenzoMinto) [reply src/favicons_covers/color.py:35]: on a second look, might be better using the condition `image_array[ :, :, 3] == 255` (0 is fully transparent, but only 255 is opaque)

File: src/favicons_covers/color.py (hunk 3 above)
Comment (LorenzoMinto) [reply src/favicons_covers/color.py]: It seems to me we are testing directly in dev. This is not really efficient. Isn't there a way to test the image processing locally? Otherwise it's very hard to iterate on solutions

Comment (aurangzaib048) [reply src/favicons_covers/color.py]: We are testing the almost 50-70 images locally, If we decided to tests the whole feed cover images locally, it will be give `403 max retries error` on most of the URL. I tried to test it locally, but I don’t think it’s wise idea but I will try it once again.

Comment (LorenzoMinto) [reply src/favicons_covers/color.py]: Is that because we are hitting the `gstatic` API for all the icons every time?

Most of the changes in the PR related to background color have been on monochromatic/transparency which is something that could be tested separately from the whole pipeline. The final stage cover images could be dumped into a folder and the bg color selection tested from there, without having to spam the APIs on each run. Otherwise iterations are too long if we just "deploy and see". 

Another approach, although more limited, could be using tests: download some representative examples and test those functions with them.

File: src/favicons_covers/color.py (hunk 2 above)
Comment (aurangzaib048) [reply src/favicons_covers/color.py:35]: I have tested using `opaque` but the results are not quite good, so I decided to keep the `fully transparent` condition.

**Review by LorenzoMinto — APPROVED** [review]
LGTM. Although in the last iteration I'm seeing the following cover images are missing:

- Al Jazeera
- Lonely Planet
- Pitchfork
- Yourweather.co.uk

On a similar note, it would be great if we had an always guaranteed final fallback like a card with the correct background color and the publisher initials, but maybe it's easier to do something like that directly on the browser cc @fallaciousreasoning
